pydm/widgets/drawing.py

```diff
@@ -1121,60 +1126,56 @@ def p2d(pt):
     def getPoints(self):
         return self._points
 
-    def _validator_(self, value):
+    def _validator(self, value):
         """
         ensure that `value` has correct form
 
         Parameters
         ----------
-        value : [str]
+        value : [ordered pairs]
             List of strings representing ordered pairs
             of integer coordinates.  Each ordered pair
-            is comma-separated (although white-space
-            separated is acceptable as input).
+            is a tuple or list.
 
         Returns
         ----------
-        verified : [str]
-            List of strings in standard format
+        verified : [ordered pairs]
+            List of `tuple(number, number)`.
 
         """
-        def isinteger(value):
-            value = value.strip()
+        def isfloat(value):
+            if isinstance(value, str):
+                value = value.strip()
             try:
                 float(value)
                 return True
             except:
                 return False
 
         verified = []
-        for i, pt in enumerate(value):
-            point = pt.split(",")
-            if len(point) != 2:
-                point = pt.split()    # tolerant of space-separated
+        for i, pt in enumerate(value, start=1):
+            point = pt
+            if isinstance(point, str):
+                point = list(map(float, point.split(",")))
+                # logger.exception("polyline point %d must be list of (number, number) pairs, received '%s'", i, pt, exc_info=False)
+                # return
             if len(point) != 2:
-                emsg = "polyline point %d must be two values, comma-separated, received '%s'" % (i+1, pt)
-                logger.exception(emsg)
-                return
-            if not isinteger(point[0]):
-                emsg = "polyline point %d content must be integer, received '%s'" % (i+1, point[0])
-                logger.exception(emsg)
+                logger.exception("polyline point %d must be two numbers, comma-separated, received '%s'", i, pt, exc_info=False)
```

Comment:
`logger.exception(msg, exc_info=False)` is equivalent to `logger.error(msg)`, it's probably better practice to simplify all of these if we don't want the `exc_info` (not that there is any exception info in this block, it's not in a try/except... weird choice prior to this PR)

Reply:
_op cit_